### ISSUE TITLE
sumologicschemaprocessor: transform attributes into JSON-like structure

### ISSUE THREAD

**Issue by aboguszewski-sumo** [SumoLogic/sumologic-otel-collector#865]
## Problem
The processor should gain possibility to transform set of attributes with common namespace prefix in the attribute name, to a nested key-value structure. 

## Example

The following list of attributes:
```
[
  "kubernetes.container_name": "xyz",
  "kubernetes.host.name": "the host",
  "kubernetes.host.address": "127.0.0.1",
  "kubernetes.namespace_name": "sumologic",
  "another_attr": "42" 
]
```
Should be translated into such list:
```
[
  "kubernetes": {
    "container_name": "xyz",
    "namespace_name": "sumologic",
    "host": {
      "name": "the host",
      "address": "127.0.0.1"
    }
  },
 "another_attr": "42"
]
```

## Notes

Is is assumed, that by "prefix", we mean the namespace path separated by `.`, for example `xyz` is a common prefix path for `xyz.abc.qwe` and `xyz.foo`, but `sumo` **is not** a prefix for `sumologic.foo.baz` and `sumologic.bar.baz`.

Optional config options that are not necessary, but might be useful:

- changing the separator from `.` to any string
- setting the translation on/off for both resource attributes and regular attributes
- denylist of prefixes that shouldn't be transformed

_Original upstream issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14977_

**Comment by aboguszewski-sumo** [comment]
@rupeshnemade there is one undefined case in this issue. Given the following set of attributes:
```
{
  "a.foo.bar": "baz",
  "a.sumo.logic": "schema"
}
```
(there is only one key in namespace `a.foo` and only one key in namespace `a.sumo`)
What should be the result? It can be either:
1.
```
{
  "a": {
    "foo.bar": "baz",
    "sumo.logic": "schema"
  }
}
```
or 2.:
```
{
  "a": {
    "foo": {
      "bar": "baz"
    },
    "sumo": {
      "logic": "schema"
    }
  }
}
```

We would definitely want to use 2., because we want to keep this feature as simple as possible, but because this is needed to reduce log size, maybe the first one makes more sense.

**Comment by rupeshnemade** [comment]
@aboguszewski-sumo For keeping lesser log size with aesthetic JSON view, first approach sounds better. 
In 1st approach, I hope it will keep that flattened format for 3rd level prefix only. 
Example, if I have multiple key-values inside kuberenets.labels.<>, then only for labels it will be flattened.

**Comment by aboguszewski-sumo** [comment]
Fine. 2nd approach has already been merged, the first one will be added soon.

**Comment by aboguszewski-sumo** [comment]
This has been resolved by #877 #880 #881
Hypothetically, this feature could be enabled only for resource attributes - this is a potential enhancement.
I will close the issue in January.
